### PR TITLE
[full-ci] Update deps and move configs to packages

### PR DESCRIPTION
**Pull request by dschmidt**
## Description
This PR updates a bunch of tools and moves their configs to packages, so we can publish them and use them from extensions at some point. 
This supersedes https://github.com/owncloud/web/pull/6715 
## Related Issue

- Supersedes https://github.com/owncloud/web/pull/6715

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oCIS</strong> https://drone.owncloud.com/owncloud/web/28957/12/1
:boom: The e2e-ocis tests pipeline failed. The build has been cancelled.

**Review by fschade — APPROVED** [review]
pretty cool 🥳 one smallish thing which is more a personal preference and 100% optional.

i like it to have the folders structured/clustered inside mono repos, maybe you too!? ;=
 
* packages/babel-presets => packages/env-babel-presets
 * packages/eslint-config => packages/env-eslint-config
 * packages/prettier-config => packages/env-prettier-config
 * packages/tsconfig => packages/env-tsconfig

up to you 😘

**Comment by pascalwengerter** [comment]
Wait this is really happening!? 😼

**Comment by dschmidt** [comment]
> pretty cool 🥳 one smallish thing which is more a personal preference and 100% optional.
> 
> i like it to have the folders structured/clustered inside mono repos, maybe you too!? ;=
> 
> * packages/babel-presets => packages/env-babel-presets
> * packages/eslint-config => packages/env-eslint-config
> * packages/prettier-config => packages/env-prettier-config
> * packages/tsconfig => packages/env-tsconfig
> 
> up to you 😘

Yeah, kinda like it too. Will play around with it, but right now I'm tempted to do that in a follow up, as I'm happy I finally got this green now 😅
I would like to update eslint soon anyway, could tackle it then. 
FTR: we would just change the folder name but keep the package name. 

> Wait this is really happening!? 😼

Yeah, I can't believe myself I got this 🟢 😅

File: packages/web-app-files/src/composables/sort/useSort.ts [hunk 1]
```diff
@@ -94,7 +94,7 @@ function createSortDirQueryRef<T>(options: SortOptions<T>): Ref<QueryValue> {
   return useRouteQueryPersisted({
     name: unref(options.sortDirQueryName) || SortConstants.sortDirQueryName,
     defaultValue: unref(firstSortableField(unref(options.fields))?.sortDir),
-    routeName: unref(options.routeName || useRouteName())
+    routeName: unref(options.routeName) || unref(useRouteName())
```
Comment (dschmidt) [inline packages/web-app-files/src/composables/sort/useSort.ts:97]: afaict these are actual bug fixes, passing in an empty ref should trigger the fallback

**Comment by pascalwengerter** [comment]
> > pretty cool 🥳 one smallish thing which is more a personal preference and 100% optional.
> > 
> > i like it to have the folders structured/clustered inside mono repos, maybe you too!? ;=
> > 
> > * packages/babel-presets => packages/env-babel-presets
> > * packages/eslint-config => packages/env-eslint-config
> > * packages/prettier-config => packages/env-prettier-config
> > * packages/tsconfig => packages/env-tsconfig
> > 
> > up to you 😘
> 
> Yeah, kinda like it too. Will play around with it, but right now I'm tempted to do that in a follow up, as I'm happy I finally got this green now 😅
> I would like to update eslint soon anyway, could tackle it then. 
> FTR: we would just change the folder name but keep the package name. 
> 
> > Wait this is really happening!? 😼
> 
> Yeah, I can't believe myself I got this 🟢 😅

Merge & iterate, then? 🤓😏

**Comment by dschmidt** [comment]
🥳🥳🥳

Should we archive https://github.com/owncloud/web-lint-config then? Maybe even add a comment to the readme it was never used in practice

**Comment by pascalwengerter** [comment]
> 🥳🥳🥳
> 
> Should we archive https://github.com/owncloud/web-lint-config then? Maybe even add a comment to the readme it was never used in practice

Could also be deleted no? It's not like it was announced/communicated outside of the core developers and one or two PRs :)

**Pull request merged**